### DISCUSSION TITLE
Erro compiling LIS 7.4

### DISCUSSION THREAD

**Discussion by laurizioA** [NASA-LIS/LISF#1198] [Q&A] [answered]
Hello!

When I run the .compile the following error appears:

**gfortran -c -ffree-line-length-0 -O2 -fbacktrace -DSPMD -fconvert=big-endian -DHIDE_SHR_MSG -DNO_SHR_VMATH -DGFORTRAN -DLINUX -I/home/codesal/anaconda3/envs/lisf/mod -I/home/codesal/anaconda3/envs/lisf/include/ -I/home/codesal/anaconda3/envs/lisf/include/ -I/home/codesal/anaconda3/envs/lisf/include/ -I/usr/local/hdf4/include/ -I/home/codesal/anaconda3/envs/lisf/include/ -DLIS_JULES -I./ -I../core/ -I../params/gfrac/SPORTDaily/ -I../params/gfrac/VIIRSDaily/ -I../surfacemodels/land/vic.4.1.1/ -I../surfacemodels/land/vic.4.1.1/physics/ -I../surfacemodels/land/vic.4.1.2.l/ -I../surfacemodels/land/vic.4.1.2.l/physics/ -I../surfacemodels/land/awral.6.0.0/physics/ -I../surfacemodels/land/rdhm.3.5.6/ -I../surfacemodels/land/rdhm.3.5.6/sac/ -I../surfacemodels/land/rdhm.3.5.6/frz/ -I../surfacemodels/land/rdhm.3.5.6/snow17/ -I../surfacemodels/land/rdhm.3.5.6/sachtet/ ../core/LIS_DAobservationsMod.F90
../core/LIS_DAobservationsMod.F90:1327:18:

1327 | call MPI_BCAST(LIS_obs_offsets(k,:), LIS_npes, MPI_INTEGER,0, LIS_mpi_comm, ierr)
| 1
......
1822 | call MPI_BCAST(gtmp1d, LIS_rc%obs_gnc(k)*LIS_rc%obs_gnr(k), MPI_REAL,0, &
| 2
Error: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(4)/REAL(4)).
../core/LIS_DAobservationsMod.F90:1168:25:

1168 | call MPI_BCAST(LIS_obs_goffsets(k,:), LIS_npes, MPI_INTEGER,0, &
| 1
......
1822 | call MPI_BCAST(gtmp1d, LIS_rc%obs_gnc(k)*LIS_rc%obs_gnr(k), MPI_REAL,0, &
| 2
Error: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(4)/REAL(4)).
../core/LIS_DAobservationsMod.F90:1192:25:

1192 | call MPI_BCAST(LIS_obsens_goffsets(k,:), LIS_npes, MPI_INTEGER,0, &
| 1
......
1822 | call MPI_BCAST(gtmp1d, LIS_rc%obs_gnc(k)*LIS_rc%obs_gnr(k), MPI_REAL,0, &
| 2
Error: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(4)/REAL(4)).
../core/LIS_DAobservationsMod.F90:369:25:

369 | call MPI_BCAST(LIS_ooffsets(k,:), LIS_npes, MPI_INTEGER,0, &
| 1
......
1822 | call MPI_BCAST(gtmp1d, LIS_rc%obs_gnc(k)*LIS_rc%obs_gnr(k), MPI_REAL,0, &
| 2
Error: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(4)/REAL(4)).
make: *** [Makefile:140: LIS_DAobservationsMod.o] Erro 1

[ERR] Compile failed
-------------------------------------------------------------------**

The installed MPI is version 4.0.2. and my operational system is ubuntu 20.04.5 LTS.

**Comment by jvgeiger** [comment, marked as answer]
Hello,

I encountered this error recently.  I have not yet formally resolved this error, but
the workaround is to add `-fallow-argument-mismatch` to the FFLAGS variable in your lis/make/configure.lis file.

**Comment by emkemp** [comment]
Hi:

I found a similar error reported with different software using GNU compilers and OpenMPI.  See https://www.mail-archive.com/users@lists.open-mpi.org/msg34359.html Bottom line is this may be due to our logic in LISF/lis/core/LIS_mpiMod.F90

! !USES:
#if (defined SPMD)
#if (defined USE_INCLUDE_MPI)
  include 'mpif.h' 
#else
  use mpi
#endif
#endif

If your build script is defining USE_INCLUDE_MPI, it will be running the include 'mpif.h' which doesn't support overloading of arguments for the MPI subroutines.  Which GNU complains about.  If the build script does not define that symbol, then the use mpi will be compiled, and that may work.

We will test on our end, but you may want to test on your end too.

**Comment by laurizioA** [comment]
Thanks @jvgeiger e @emkemp !
